shoop/utils/enums.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+import six
+
+
+def map_enum(enum_cls, value, default=None):
+    """
+    Attempt to cast the given `value` into a value of the given `enum_cls`.
+
+    Tries by value first, then case-insensitively by name.
+
+    If casts fail, the `default` is returned.
+
+    This is useful for places where it doesn't make sense to export the actual
+    enumeration classes (such as into templates, etc.) while still avoiding
+    "magic constants" (i.e. the actual enum values).  For the aforementioned
+    reason of being used in templates, it is purposefully slightly more
+    lenient in its parsing than more "core" utilities might be.
+
+    :param enum_cls: Enum class
+    :type enum_cls: enum.Enum
+    :param value: value or name
+    :type value: object|str
+    :param default: Default value to return in case of trouble
+    :type default:
+    :return:
+    :rtype:
+    """
+    try:
+        return enum_cls(value)
+    except ValueError:
+        pass
+    members = getattr(enum_cls, "_map_enum_cache_", None)
+    if members is None:
+        members = dict((k.lower(), v) for (k, v) in enum_cls.__members__.items())
+        enum_cls._map_enum_cache_ = members
+    return members.get(six.text_type(value).lower(), default)
```

Comment:
`default` is not good keyword for a fallback value here, since it's just a way to hide ValueError of unknown enum value.

Is it really needed? I think fallbacking to None is more harmful than raising ValueError.